<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NetworkConfig</class>
 <widget class="IConfigItem" name="NetworkConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Network</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="5" column="2">
    <widget class="QPushButton" name="m_EditButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Edit</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="3">
    <widget class="QTableWidget" name="m_NodeList">
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="showDropIndicator" stdset="0">
      <bool>false</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
     <property name="cornerButtonEnabled">
      <bool>false</bool>
     </property>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>IP address</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QPushButton" name="m_AddButton">
     <property name="text">
      <string>Add</string>
     </property>
    </widget>
   </item>
   <item row="5" column="3">
    <widget class="QPushButton" name="m_RemoveButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Remove</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_Nodes">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Nodes:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="3">
    <widget class="QSpinBox" name="m_Port">
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>65535</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_Port">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>TCP port:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="4">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>IConfigItem</class>
   <extends>QWidget</extends>
   <header>IConfigItem.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_Port</tabstop>
  <tabstop>m_NodeList</tabstop>
  <tabstop>m_AddButton</tabstop>
  <tabstop>m_EditButton</tabstop>
  <tabstop>m_RemoveButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>m_NodeList</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>NetworkConfig</receiver>
   <slot>editNode(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_NodeList</sender>
   <signal>currentCellChanged(int,int,int,int)</signal>
   <receiver>NetworkConfig</receiver>
   <slot>onSelectionChanged(int,int,int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_AddButton</sender>
   <signal>clicked()</signal>
   <receiver>NetworkConfig</receiver>
   <slot>addNode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_EditButton</sender>
   <signal>clicked()</signal>
   <receiver>NetworkConfig</receiver>
   <slot>editNode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_RemoveButton</sender>
   <signal>clicked()</signal>
   <receiver>NetworkConfig</receiver>
   <slot>removeNode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_Port</sender>
   <signal>valueChanged(int)</signal>
   <receiver>NetworkConfig</receiver>
   <slot>setPort(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>18</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addNode()</slot>
  <slot>editNode()</slot>
  <slot>editNode(int,int)</slot>
  <slot>removeNode()</slot>
  <slot>onSelectionChanged(int,int,int,int)</slot>
  <slot>setPort(int)</slot>
 </slots>
</ui>
